llvm/include/llvm/Analysis/MemoryLocation.h

```diff
@@ -150,16 +159,22 @@ class LocationSize {
       return beforeOrAfterPointer();
     if (Value == AfterPointer || Other.Value == AfterPointer)
       return afterPointer();
+    if (isScalable() != Other.isScalable())
+      return afterPointer();
 
     return upperBound(std::max(getValue(), Other.getValue()));
```

Comment:
It looks like this max will implicitly convert to uint64_t. The != above for scalable vectors may need to be a ||.